### DISCUSSION TITLE
How to write custom user attributes during registration

### DISCUSSION THREAD

**Discussion by funktionierbar** [goauthentik/authentik#4767] [Q&A]
Hi everybody,

i have been wrecking my brain the whole day trying to figure this out, now i am going to ask for help:

I think the problem is simple. I want to have the user fill in certain information that is stored as a custom attribute. For Example an external mail adress, so that during password recovery the recovery link can be sent to a secondary mail, as my primary mail is behind authentik. I use invitation flows for different groups of users. My Invitation flow has these stages: Invitation, Prompt, Write, Login. 
I created the custom fields in the prompt stage, validation works, the context gets updated accordingly, but the custom fields are not written to the new user. 
I figured I need an expression policy for the write stage to write the prompt information to the pending_user model.
and thats were I got stuck. 
Best i could muster was something like this: 

>ak_logger.info(context['pending_user'])
>context['pending_user'].'mailActivated' = true
>context['pending_user'].'email-external' = request.context.get('prompt_data').get('email_external')
>return True

It always fails with "Request has been denied. Unkown error", and the logs show nothing that tells me anything, except maybe:

>{"auth_via": "unauthenticated", "event": "No Pending user to login.", "host": "auth.example.de", "level": "debug", "logger": "authentik.flows.stage", "pid": 8045, "request_id": "728c1512669941f08d6debc10c0e596a", "stage": "default-source-enrollment-login", "stage_view": "authentik.stages.user_login.stage.UserLoginStageView", "timestamp": "2023-02-23T16:13:24.054747"}
>{"auth_via": "unauthenticated", "event": "f(exec): Stage invalid", "flow_slug": "nice36-invitation-einrollment", "host": "auth.example.de", "level": "debug", "logger": "authentik.flows.views.executor", "pid": 8045, "request_id": "728c1512669941f08d6debc10c0e596a", "timestamp": "2023-02-23T16:13:24.055231"}

Via mail I get a message like this:
>Traceback (most recent call last): File "user-write-customfields", line 5, in <module> File "user-write-customfields", line 2, in handler builtins.KeyError: 'pending_user'

Can you help me out here? How do I write to the user to be created?

**Comment by marekhotshot** [comment]
had similar issue, no need for expression policy.
https://github.com/goauthentik/authentik/issues/7879

you need to add **attributes.**CustomAttributeName when creating new attribute to make it write to user.

**Reply by sburk** [reply]
Where do you place **attributes.**CustomAttributeName ?
I want to auto generate user name from email (first part before @).
Can you help me?

**Reply by partofthething** [reply]
I gotchu. Struggled through it and realized that this is what they mean: 

https://github.com/goauthentik/authentik/issues/7879#issuecomment-2813233883
